### PR TITLE
Allow assessment bonus points through manual grading over 100%

### LINKED ISSUE #9713: Allow assessment bonus points to account for question score beyond 100%
The implementation of assessment bonus points in #4090 was intended to allow some questions to effectively act as "bonus questions", by allowing the assessment score to go above 100% based on question completion (rather than credit for early work, which was already an option). This implementation allows the setting of `maxBonusPoints`, which allowed the sum of all question points to add up to more than the assessment's `maxPoints`, and thus the score to more than 100%.

This implementation, though, [limited `maxPoints+maxBonusPoints` to the total points assigned to the questions in the assessment](https://github.com/PrairieLearn/PrairieLearn/blob/0fb66e21e26508ed0a5a1f9d0a7fa2377971c047/apps/prairielearn/src/sprocs/assessment_instances_update.sql#L108). This assumes that questions are limited to a score of 100%, which is true in most auto-graded contexts, but can be overridden in manual grading questions. A manual grading can provide points that go above the limit of the question. In fact, the rubric settings even have an explicit option to override the maximum number of points in the question.

The example proposed by @firasm was an assessment with a single question that had 20 auto points and 80 manual points, with a rubric on manual points that allowed a question to get up to 82 points. The meant that the question could go up to 102 (out of 100) points. However, since the question is assigned 100 points, this would limit the assessment score to 100. Setting `maxBonusPoints` to 2, with `maxPoints` at 100, would be overridden by the assessment instance update sproc to set `maxBonusPoints` to zero, so the setting would be moot.

The proposal in this issue (as discussed on Slack) is that this limit on `maxBonusPoints` be removed, allowing the assessment to honour the bonus points assigned in manual grading in the assessment score. This would only affect assessments that set both `maxPoints` and `maxBonusPoints` to a specific value, since setting `maxBonusPoints` without `maxPoints` [has a well-defined behaviour that should remain for backwards compatibility](https://github.com/PrairieLearn/PrairieLearn/blob/0fb66e21e26508ed0a5a1f9d0a7fa2377971c047/docs/assessment.md?plain=1#L154). In essence, this would mean removing the `LEAST` restriction in this line:
https://github.com/PrairieLearn/PrairieLearn/blob/0fb66e21e26508ed0a5a1f9d0a7fa2377971c047/apps/prairielearn/src/sprocs/assessment_instances_update.sql#L108
plus updates in documentation.

### PR DESCRIPTION
**Pull request by jonatanschroeder**
Resolves #9713.

### CODE REVIEW

**Comment by mwest1066** [comment]
How does this interact with #8890 and/or the various [question point/percentage changes](https://github.com/PrairieLearn/PrairieLearn/issues/8858#issuecomment-1828208920) that are coming?

File: apps/prairielearn/src/sprocs/assessment_instances_update.sql [hunk 1]
```diff
@@ -33,7 +33,7 @@ BEGIN
     SELECT
         c.id,      g.id,   u.user_id,            a.id,          a.type,
         a.max_points,          ai.max_points,
-        COALESCE(a.max_bonus_points, 0),    ai.max_bonus_points,
+        GREATEST(a.max_bonus_points, 0),    ai.max_bonus_points,
```
Comment (mwest1066) [inline apps/prairielearn/src/sprocs/assessment_instances_update.sql:36]: Is this change simply to guard against the fact that a user might set `assessment.max_bonus_points` to a negative value?

**Review by mwest1066 — APPROVED** [review]
I'm fine with this change, although I think that it might be even nicer to fix #9714 by only applying `max_bonus_points` to auto points, and always allowing manual points to exceed 100% of the allocated manual points. With the change in this PR, an instructor will have to anticipate that they will be awarding bonus manual points and explicitly allow that (with a precise upper limit) in the assessment, which feels error prone to me.

All that being said, the current PR is simple and clearly an improvement over the current state, so we should proceed with it.

File: apps/prairielearn/src/sprocs/assessment_instances_update.sql (hunk 1 above)
Comment (jonatanschroeder) [reply apps/prairielearn/src/sprocs/assessment_instances_update.sql:36]: Indeed, though this `GREATEST` logic was already in place in the removed line below, I just moved the logic to the query instead.

Comment (mwest1066) [reply apps/prairielearn/src/sprocs/assessment_instances_update.sql:36]: Yep, seems reasonable.

**Pull request merged**